### DISCUSSION TITLE
get-key Query

### DISCUSSION THREAD

**Discussion by MailNSrinivasan** [schibsted/jslt#334] [Q&A]
Hi, I need some help on the below while using get-key function

```
let node = {
  "aa": "11",
  "bb" : "22",
  "cc" : {
     "cc1": "33"
  }
}

let path = "cc.cc1"

{
   "test" : get-key($node, $path)
}
```

In the above code I am not able to get the value using a path like "cc.cc1" while using the get-key method. Is there any way I can achieve this. I can only get the path dynamically. 

Thanks
Srini

**Comment by catull** [comment]
You have to do it in 2 steps.

```
let node = {
  "aa": "11",
  "bb" : "22",
  "cc" : {
     "cc1": "33"
  }
}

let path = "cc"
let cc = get-key ($node, $path)

{
   "test" : $cc.cc1
}
```

**Reply by MailNSrinivasan** [reply]
But I will know the path "xx.yy" at runtime only.

**Reply by catull** [reply]
Still, you can only use simple keys, not combined ones.
Good luck.

**Comment by catull** [comment]
Another way, in this case.

Say the input is:

```json
{
  "aa": "11",
  "bb" : "22",
  "cc" : {
     "cc1": "33"
  }
}
```

Then you can do this transformation:

```
{
   "test" : .cc.cc1
}
```

**Reply by MailNSrinivasan** [reply]
okay let me explain it better. the path ".cc.cc1" can change at runtime. I dont know the path while I write the JSLT. It will be the value somewhere and I get it only at runtime. it could be ".bb.bb1" or any thing else. Thats the reason I want to use get-key function. Let me know is that clear :)

**Reply by catull** [reply]
.get-key is really useful for hash-map lookups, single-key must be used.
You can write JSTL that accepts parameters, such as

```
process (node, dynamicPath) {
  "test": $node.$dynamicPath
}
```

Does this apply to your use case ?

**Reply by MailNSrinivasan** [reply]
I tried the above $node.$dynamicPath in demo play ground, it throws error.

**Reply by catull** [reply]
You are giving us very little, can't help you.
